src/bytes_mut.rs

```diff
@@ -636,11 +663,25 @@ impl BytesMut {
                 // sure that the vector has enough capacity.
                 let v = &mut (*shared).vec;
 
-                if v.capacity() >= new_cap {
-                    // The capacity is sufficient, reclaim the buffer
-                    let ptr = v.as_mut_ptr();
+                let v_capacity = v.capacity();
+                let ptr = v.as_mut_ptr();
+
+                // The following line is equivalent to:
+                //
+                //     self.ptr.as_ptr().offset_from(ptr) as usize;
+                //
+                // But due to min rust is 1.39 and it is only stablised
+                // in 1.47, we cannot use it.
+                let offset = self.ptr.as_ptr() as usize - ptr as usize;
```

Comment:
In bytes, we usually put this kind of functionality in helper functions. Can you move this into an `offset_from` helper fn? You can put it next to the `invalid_ptr` helper fn.

Reply:
Sure

Reply:
Done!